server/mailet/integration-testing/src/test/java/org/apache/james/mailets/ForwardIntegrationTest.java

```diff
@@ -323,19 +323,13 @@ void forwardShouldSplitLocalCopy() throws Exception {
             .authenticate(FROM, PASSWORD)
             .sendMessage(FROM, BOB.asString());
 
-        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 2L);
+        Awaitility.await().until(() -> mailRepositoryProbe.getRepositoryMailCount(CUSTOM_REPOSITORY) == 1L);
 
         SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
             Mail forwardedMail = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
-                .filter(Throwing.predicate(mail -> mail.getRecipients().contains(DELPHINE.asMailAddress())))
                 .findAny().get();
-            Mail localCopy = mailRepositoryProbe.listMails(CUSTOM_REPOSITORY)
-                .filter(Throwing.predicate(mail -> mail.getRecipients().contains(BOB.asMailAddress())))
-                .findAny().get();
-            softly.assertThat(forwardedMail.getRecipients()).containsOnly(DELPHINE.asMailAddress());
+            softly.assertThat(forwardedMail.getRecipients()).containsOnly(DELPHINE.asMailAddress(), BOB.asMailAddress());
             softly.assertThat(forwardedMail.getMaybeSender().asOptional()).contains(BOB.asMailAddress());
-            softly.assertThat(localCopy.getRecipients()).containsOnly(BOB.asMailAddress());
-            softly.assertThat(localCopy.getMaybeSender().asOptional()).contains(new MailAddress(FROM));
         }));
```

Comment:
Because now local copy is handled in the forwarded mail. 
This ease the overall logic a lot, and the end result is the same (message in both INBOX)

Reply:
two concerns:
- when Bob see the mail, the sender is him. does it make sense?
- why doesn't mailRepositoryProbe have two identical mails? (because the mail is forwarded to Bob and Delphine)

Reply:
> when Bob see the mail, the sender is him. does it make sense?

You confuse envelope and headers